### ISSUE TITLE
Fallback language translations failing to be processed.

### ISSUE THREAD

**Issue by xMonkfish** [angular-translate/angular-translate#1368]
I have angular translate configured using the following code.

```javascript 
        $translateProvider.useStaticFilesLoader({
            files: [{
                prefix: 'i18n/sys-captions.',
                suffix: '.json'
            }]
        }).fallbackLanguage('en_US')
        .determinePreferredLanguage();
```

In this instance determinePreferredLanguage correctly sets my preferred language as en_GB (I'm currently experimenting with how reliable it is). I only have the US translation file available and the fallbackLanguage file is correctly loaded from the server.

However, when I try and translate an item via the $translate service, it will fail to find the item. The problem seems to reside inside the determineTranslation function.

```javascript
if ($uses && $fallbackLanguage && $fallbackLanguage.length) {
   fallbackTranslation(translationId, interpolateParams, Interpolator, defaultTranslationText)
```

In my cases the $uses variable appears as undefined and so it does not make an attempt to process the fallback translation. Removing the $uses variable from the if statement corrects the issue for me.

**Comment by tspaeth** [comment]
Hi! Can you setup a short plunkr wit a complete example? As this "bug" comes up from time to time out in the user base, I'm now just collecting all the ways people setup fallbacks etc. and then find a solution that will "cover" all scenarios if possible. thanks.

**Comment by xMonkfish** [comment]
Here is one I've set up below - it's not failing for precisely the same reason as when I run locally though. 

https://plnkr.co/edit/UG9NPZDmxF29sw44cxjU

In this case it appears to be failing because it waits for an asynchronous load which fails (presumably to fetch the en_GB locale although I've not confirmed.)

The Chrome Network tab does show the en_GB file failing, but a successful call to fetch the fallback en_US language. The sample does work if your language is set to en_US.

**Comment by xMonkfish** [comment]
Ignore the above. I noticed the plunkr had used an old version of the library. Changing it from 2.6.0 to 2.8.1 means it can't be recreated there. I will make sure that this isn't the reason it's failing in production too. We use Visual Studio's version prompts for our bower file. So not sure if that is what has happened here too.

**Comment by xMonkfish** [comment]
I've tried again this morning on our production code and can't get it to fail there anymore either, so I'll close this issue. I'll see if anyone else has the problem internally as I believe it may be related to if things happen to load in a certain order. Then again I may have just been unlucky.

Thanks for your time.

**Comment by xMonkfish** [comment]
Hi again.

I've managed to reproduce this issue on Plunker now (along with a work around)

Turns out the fault is caused if you call fallbackLanguage before determinePreferredLanguage while using an async loader.

If you swap the function calls around then it works as expected. https://plnkr.co/edit/r7YNxfMPBfVaXnLVm4ak?p=preview

**Comment by tspaeth** [comment]
Yes, this behavior is already a known "bug" but I instantly can't find the corresponding ticket number...

**Comment by envision** [comment]
We are experiencing this bug also

**Comment by envision** [comment]
When using .determinePreferredLanguage().fallbackLanguage('en_US') English doesn't get applied. But when using determinePreferredLanguage() only, then my browser's language Finnish gets applied.

**Comment by envision** [comment]
Even this logs twice "undefined":

    console.log($translate.use());
    $translate.use('fr_FR');
    console.log($translate.use());

**Comment by knalli** [comment]
@tspaeth  Can you handle this one?

**Comment by tspaeth** [comment]
Yup, picking that.

**Comment by envision** [comment]
:+1:

**Comment by tspaeth** [comment]
@envision do you have the possibility to check if the PR (even though it might not be merged yet) will fix your issue (against canary probably) thanks.

**Comment by envision** [comment]
Sure do @tspaeth – I have now verified this fix worked for me, thanks a lot! Fallback language's translation string is now loaded if target language doesn't have it translated.

**Comment by knalli** [comment]
#1433  is landed in canary. @envision  Thank you for verifying.

Available in the next 2.11
